app/scripts/views/sign_up.js

```diff
@@ -218,6 +223,27 @@ function (Cocktail, _, p, BaseView, FormView, Template, AuthErrors, mailcheck,
       return this._isMailcheckEnabledValue;
     },
 
+    passwordCheck: function () {
+      var self = this;
+      var password = this.getElementValue('.password');
+      try{
+        this.passwordcheck(password, function (msg) {
+          if (msg === 'MISSING_PASSWORD') {
+            // same as PASSWORD_REQUIRED
+            msg = self.translateError(AuthErrors.toError('PASSWORD_REQUIRED'));
+            self.logError(msg);
+          } else if (AuthErrors.ERRORS[msg] !== undefined) {
+            msg = self.translateError(AuthErrors.toError(msg));
+            self.logError(msg);
+          }
+          // in the future, do some fancy tooltip here.
+        });
+      } catch(err) {
```

Comment:
@shane-tomlinson It doesnt throw an exception right now, but I added this just to cover all the error cases.